Floofbot/Modules/Logging.cs

```diff
@@ -20,79 +20,123 @@ public class Logging
+                        UserMutedChannel = 0,
+                        UserUnmutedChannel = 0,
+                        IsOn = false
+                    });
                     _floofDB.SaveChanges();
+                }
+            }
 
+            private bool TryLinkChannelType(string channelType, Discord.IChannel channel, Discord.IGuild guild)
+            {
+                ulong serverId = guild.Id;
+                ulong channelId = channel == null ? 0 : channel.Id;
+                AddServerIfNotExists(serverId);
+                try
+                {
+                    string updateCommand = $"UPDATE LogConfigs SET {channelType} = {channelId} WHERE ServerID = {serverId}";
+                    _floofDB.Database.ExecuteSqlRaw(updateCommand);
+                    _floofDB.SaveChanges();
+                    return true;
+                }
+                catch (Exception e)
+                {
+                    string errorMsg = $"Error: Unable to link {channelType} to <#{channelId}>";
+                    Log.Error(errorMsg + Environment.NewLine + e);
+                    return false;
                 }
             }
 
-            private async Task SetChannel(string tableName, Discord.IChannel channel, Discord.IGuild guild)
+            [Summary("Links or re-links a channel type onto a single channel in a server")]
+            [Command("setchannel")]
+            public async Task SetChannelType(
+                [Summary("channel type")] string channelType = null,
+                [Summary("channel (current channel if unspecified)")] Discord.IChannel channel = null)
             {
-                CheckServer(guild.Id);
+                if (channel == null)
+                {
+                    channel = (IChannel)Context.Channel;
+                }
 
-                // set channel
-                _floofDB.Database.ExecuteSqlRaw($"UPDATE LogConfigs SET {tableName} = {channel.Id} WHERE ServerID = {guild.Id}");
-                _floofDB.SaveChanges();
-                await Context.Channel.SendMessageAsync("Channel updated! Set " + tableName + " to <#" + channel.Id + ">");
+                if (CHANNEL_TYPES.Contains(channelType))
+                {
+                    if (TryLinkChannelType(channelType, channel, Context.Guild))
+                    {
+                        await Context.Channel.SendMessageAsync($"Channel updated! Set {channelType} to <#{channel.Id}>");
+                    }
+                    else
+                    {
+                        await Context.Channel.SendMessageAsync($"Unable to set {channelType} to <#{channel.Id}>");
+                    }
+                }
+                else
+                {
+                    EmbedBuilder builder = new EmbedBuilder()
+                    {
+                        Description = $"💾 Accepted channel types: ```{string.Join(", ", CHANNEL_TYPES)}```",
+                        Color = Color.Magenta

```

Comment:
Should colour be a constant?

Reply:
For now, yeah
If we're randomizing the colour, I'd rather have a discussion on which parts should be randomized first.